Add A.S. funded food field to event form

SheepTester opened the pull request:
<!-- Fill in all spots surrounded by brackets and confirm all check boxes after confirming completion of the tasks -->

# Info



Depends on acmucsd/membership-portal#456

<!-- If there is no issue for this pull request yet, please create one or
delete this line if the pull request is for a very minor tweak. -->

This allows the marketing team to populate the food field for events. `foodItems` isn't used anywhere else in the UI yet, but letting it be populated helps to avoid blocking generating AS attendance QR codes to the Discord bot.

<!-- What changes did you make? List all distinct problems that this PR addresses. Explain any relevant
motivation or context. -->

## Changes

- Update types in apiRequests and apiResponses
- Add a field to the create and edit events form

# Type of Change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
      expected)
- [ ] Logistics Change (A change to a README, description, or dev workflow setup like
      linting/formatting)
- [ ] Continuous Integration Change (Related to deployment steps or continuous integration
      workflows)
- [ ] Other: (Fill In) <!-- Edit this type of change if you select this -->

# Testing

I have tested that my changes fully resolve the linked issue ...

- [ ] locally on Desktop.
- [x] on the live deployment preview on Desktop.
- [ ] on the live deployment preview on Mobile.
- [ ] I have added new Cypress tests that are passing.

# Checklist

- [x] I have performed a self-review of my own code.
- [x] I have followed the style guidelines of this project.
- [x] I have documented any new functions in `/src/lib/*` and commented hard to understand areas
      anywhere else.
- [x] My changes produce no new warnings.

# Screenshots

<!-- If you made any visual changes to the website, please include relevant screenshots below. -->
![{DC84B23C-7039-4ABC-9F3D-D02B8FE85109}](https://github.com/user-attachments/assets/487d91c7-852b-40bb-a1a6-3a380576be9b)

SheepTester committed:
Add A.S. funded food field to event form

newracket commented on src/lib/types/apiResponses.ts:
```diff
@@ -133,6 +133,7 @@ export interface PublicEvent {
   requiresStaff: boolean;
   staffPointBonus: number;
   discordEvent: string | null;
+  foodItems: string;
```
The foodItems field is nullable, so I think this should be `string | null`?

SheepTester replied:
```suggestion
  foodItems: string | null;
```

newracket reviewed (changes requested):
Looks pretty good to me, just one small change

SheepTester committed:
Make food items nullable

SheepTester commented on src/lib/types/apiRequests.ts:
```diff
@@ -165,6 +165,7 @@ export interface OptionalEventProperties {
   requiresStaff?: boolean;
   staffPointBonus?: number;
   discordEvent: string | null;
+  foodItems?: string;
```
```suggestion
  foodItems?: string | null;
```

SheepTester committed:
Make food items nullable in apiRequests.ts as well

Mostly out of laIziness (idk if backend truly would enjoy receiving a null) but it's in line with Discord events

newracket reviewed (approved):
LGTM
